qt/scientific_interfaces/General/MantidEV.cpp

```diff
@@ -1032,6 +1051,17 @@ void MantidEV::changeHKL_slot() {
   std::string event_ws_name =
       m_uiForm.SelectEventWorkspace_ledt->text().trimmed().toStdString();
   worker->copyLattice(peaks_ws_name, md_ws_name, event_ws_name);
+
+  bool create_hkl_workspace = m_uiForm.CreateHKLWorkspace_ckbx->isChecked();
+  if (create_hkl_workspace) { // Redo the HKL md_workspace
```

Comment:
Don't want to redo this if changeHKL fails (e.g. you give a bad transformation matrix):

if (create_hkl_workspace & **<changeHKL was succesful>**) {